### ISSUE TITLE
Confusing "Could not resolve module" errors when verifying types of a sub-package.

### ISSUE THREAD

**Issue by mjpieters** [microsoft/pyright#9529]
I tried to use the `--verifytypes` option on a submodule of a Python project. This works, except that the *Public modules* section contains a series of errors that indicates that pyright tried to resolve _other_ submodules of the same project against the selected submodule.

Concretely, the project in question is `prefect`, where I wanted to validate their public SDK package `prefect.main`; the command line used is `pyright --verifytypes prefect.main`, and the resulting public modules section output is:

```
Public modules: 1
   prefect.main
   error: Could not resolve module "prefect.main.blocks"
   error: Could not resolve module "prefect.main.cli"
   error: Could not resolve module "prefect.main.cli.cloud"
   error: Could not resolve module "prefect.main.client"
   error: Could not resolve module "prefect.main.client.schemas"
   error: Could not resolve module "prefect.main.client.types"
   error: Could not resolve module "prefect.main.concurrency"
   error: Could not resolve module "prefect.main.concurrency.v1"
   error: Could not resolve module "prefect.main.deployments"
   error: Could not resolve module "prefect.main.deployments.steps"
   error: Could not resolve module "prefect.main.docker"
   error: Could not resolve module "prefect.main.events"
   error: Could not resolve module "prefect.main.events.cli"
   error: Could not resolve module "prefect.main.events.schemas"
   error: Could not resolve module "prefect.main.infrastructure"
   error: Could not resolve module "prefect.main.infrastructure.provisioners"
   error: Could not resolve module "prefect.main.input"
   error: Could not resolve module "prefect.main.locking"
   error: Could not resolve module "prefect.main.logging"
   error: Could not resolve module "prefect.main.records"
   error: Could not resolve module "prefect.main.runner"
   error: Could not resolve module "prefect.main.runtime"
   error: Could not resolve module "prefect.main.server"
   error: Could not resolve module "prefect.main.server.api"
   error: Could not resolve module "prefect.main.server.api.ui"
   error: Could not resolve module "prefect.main.server.database"
   error: Could not resolve module "prefect.main.server.events"
   error: Could not resolve module "prefect.main.server.events.models"
   error: Could not resolve module "prefect.main.server.events.schemas"
   error: Could not resolve module "prefect.main.server.events.services"
   error: Could not resolve module "prefect.main.server.events.storage"
   error: Could not resolve module "prefect.main.server.models"
   error: Could not resolve module "prefect.main.server.orchestration"
   error: Could not resolve module "prefect.main.server.schemas"
   error: Could not resolve module "prefect.main.server.services"
   error: Could not resolve module "prefect.main.server.utilities"
   error: Could not resolve module "prefect.main.server.utilities.messaging"
   error: Could not resolve module "prefect.main.server.utilities.schemas"
   error: Could not resolve module "prefect.main.settings"
   error: Could not resolve module "prefect.main.settings.models"
   error: Could not resolve module "prefect.main.settings.models.server"
   error: Could not resolve module "prefect.main.telemetry"
   error: Could not resolve module "prefect.main.testing"
   error: Could not resolve module "prefect.main.testing.standard_test_suites"
   error: Could not resolve module "prefect.main.types"
   error: Could not resolve module "prefect.main.utilities"
   error: Could not resolve module "prefect.main.utilities.schema_tools"
   error: Could not resolve module "prefect.main.workers"
```

The rest of the report is still extremely useful and as far as I am aware, correct.

Reproduce with:

```
python -m venv prefect-pyright-verifytypes
./prefect-pyright-verifytypes/bin/pip install pyright prefect
./prefect-pyright-verifytypes/bin/pyright --verifytypes prefect.main
```

#### Context

- pyright version: 1.1.389
- environment: macos 14.7.1 (23H222), arm64
- python version: 3.12.3

**Comment by erictraut** [comment]
The `--verifytypes` command is intended to be run on an installed library. It appears that you've attempted to run the command on the repo for the library rather than in its installed form.

I tried installing prefect using `pip install prefect`. Then I ran `pyright --verifytypes prefect`, and it seemed to work fine for me. I don't see any of the above errors.

I'm going to close the issue because I think that pyright is working correctly. If I misunderstood the issue or missed something, let me know.

**Comment by mjpieters** [comment]
> I tried installing prefect using pip install prefect. Then I ran pyright --verifytypes prefect, and it seemed to work fine for me. I don't see any of the above errors.

That's not the right reproducer.

I didn't run the command in the repository, I used `pip install -e` on the repository and ran pyright on the installed **`prefect.main` package**, so a specific package *inside* the `prefect` library.

Perhaps that's not supposed to work either but let's make that explicit if so.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.391.
